File: day1.rb
def get_values(words = false)
	sum = 0;
	File.open("day1Input.txt") do |f|
		f.each_line do |line|
			if words
				## Keeping first and last letter of each word avoids issue with overlap e.g. "eightwo" would become "eigh2" when correct first digit is 8
				## Feels a bit hacky, but saves iterating through the line letter by letter
				wordsHash = {"one" => "o1e", "two" => "t2o", "three" => "t3e", "four" => "f4r", "five" => "f5e", "six" => "s6x", "seven" => "s7n", "eight" => "e8t", "nine" => "n9e", "zero" => "z0o"}
				wordsHash.each {|k,v| line.gsub!(k,v)}
			end
			line.gsub!(/[a-zA-Z\n]+/, '')
			value = Integer(line[-1]) + (Integer(line[0]) * 10)
			sum += value
		end
	end
	return sum
end

puts get_values
puts get_values(true)

File: day01/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func print_error(err error) {
	if err != nil {
		fmt.Println(err)
	}
}

func get_values(words bool) {
	file, err := os.Open("day1Input.txt")
	print_error(err)
	defer file.Close()

	totalValues := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		if words {
			wordsMap := map[string]string{"one": "o1e", "two": "t2o", "three": "t3e", "four": "f4r", "five": "f5e", "six": "s6x", "seven": "s7n", "eight": "e8t", "nine": "n9e", "zero": "z0o"}
			for key, value := range wordsMap {
				line = strings.ReplaceAll(line, key, value)
			}
		}
		digitsOnly := func(r rune) rune {
			if r >= '0' && r <= '9' {
				return r
			}
			return -1
		}
		line = strings.Map(digitsOnly, line)
		valueStr := line[:1] + line[len(line)-1:]
		value, err := strconv.Atoi(valueStr)
		print_error(err)
		totalValues += value
	}
	fmt.Println(totalValues)
}

func main() {
	get_values(false)
	get_values(true)
}
